pkg/agent/runner.go

```diff
@@ -1373,16 +1373,16 @@ func (s *atomicUpdateState) computeUnitsBounds() (uint16, uint16) {
 //
 // This method does not respect any bounds on Compute Units placed by the VM's maximum or minimum
 // resource allocation.
-func (s *atomicUpdateState) requiredCUForRequestedUpscaling() uint16 {
-	var required uint16
+func (s *atomicUpdateState) requiredCUForRequestedUpscaling() uint32 {
+	var required uint32
 
 	// note: floor(x / M) + 1 gives the minimum integer value greater than x / M.
 
 	if s.requestedUpscale.Cpu {
-		required = util.Max(required, s.vm.Cpu.Use/s.computeUnit.VCPU+1)
+		required = util.Max(required, uint32(s.vm.Cpu.Use/s.computeUnit.VCPU)+1)
 	}
 	if s.requestedUpscale.Memory {
-		required = util.Max(required, s.vm.Mem.Use/s.computeUnit.Mem+1)
+		required = util.Max(required, uint32(s.vm.Mem.Use/s.computeUnit.Mem+1))
```

Comment:
The difference in positioning of the parens at the end of these lines looks incorrect (I know it's actually fine, but best to keep them consistent).

I'd change the second one to match the first (i.e. `uint32(...)+1`)